web/pgadmin/browser/static/js/new_menus.js

```diff
@@ -0,0 +1,236 @@
+      }
+      ctxIndex++;
+    });
+
+    return this.ctxMenus;
+  }
+}
+
+
+class MenuItem {
+  constructor(options) {
+    let menu_opts = [
+      'name', 'label', 'priority', 'module', 'callback', 'data', 'enable',
+      'category', 'target', 'url' /* Do not show icon in the menus, 'icon' */, 'node',
+      'checked', 'below', 'menu_items', 'is_checkbox', 'action', 'applies', 'is_native_only'
+    ];
+    let defaults = {
+      url: '#',
+      target: '_self',
+      enable: true,
+    };
+    _.extend(this, defaults, _.pick(options, menu_opts));
+    if (!this.callback) {
+      this.callback = (item) => {
+        if (item.url != '#') {
+          window.open(item.url);
+        }
+      };
+    }
+  }
+
+  contextMenuCallback(self, item) {
+    let o = self,
+      cb;
+
+    if (o.module['callbacks'] && (
+      o.callback in o.module['callbacks']
+    )) {
+      cb = o.module['callbacks'][o.callback];
+    } else if (o.callback in o.module) {
+      cb = o.module[o.callback];
+    }
+    if (cb) {
+      cb.apply(o.module, [o.data, item]);
+    } else {
+      pgAdmin.Browser.report_error(
+        pgadminUtils.sprintf('Developer Warning: Callback - "%s" not found!', o.cb)
+      );
+    }
+  }
+
+  getContextItem(label, is_disabled, item, sub_ctx_item) {
+    let self = this;
+    return {
+      name: label,
+      /* icon: this.icon || this.module && (this.module.type), */
+      disabled: is_disabled,
+      callback: () => { this.contextMenuCallback(self, item); },
+      ...(sub_ctx_item && Object.keys(sub_ctx_item).length > 0) && { items: sub_ctx_item }
+    };
+  }
+
+
+  change_checked() {
+
+  }
+
+  /*
+     * Updates the enable/disable state of the menu-item based on the current
+     * selection using the disabled function. This also creates a object
+     * for this menu, which can be used in the context-menu.
+     */
+  update(node, item, menuItem, manuGroup) {
+    this.is_disabled = this.disabled(node, item);
+    pgAdmin.Browser.Events.trigger('nw-enable-disable-menus', pgAdmin.Browser, menuItem, manuGroup);
+  }
+
+  /*
+     * Checks this menu enable/disable state based on the selection.
+     */
+  disabled(node, item) {
+    if (this.enable == undefined) {
+      return false;
+    }
+
+    if (this.node) {
+      if (!node) {
+        return true;
+      }
+      if (_.isArray(this.node) ? (
+        _.indexOf(this.node, node) == -1
+      ) : (this.node != node._type)) {
+        return true;
+      }
+    }
+
+    if (_.isBoolean(this.enable)) return !this.enable;
+    if (_.isFunction(this.enable)) {
+      return !this.enable.apply(this.module, [node, item, this.data]);
+    }
+    if (this.module && _.isBoolean(this.module[this.enable])) {
+      return !this.module[this.enable];
+    }
+    if (this.module && _.isFunction(this.module[this.enable])) {
+      return !(this.module[this.enable]).apply(this.module, [node, item, this.data]);
+    }
+
+    return false;
+  }
+}
+
+
+function createMenu() {
```

Comment:
Move to this Menu class with name `static function create`.
And return the menuObj as return value.

Caller can call as `const menuObj = Menu.create()`

Reply:
Done